File: c#/Sort/BubbleSort.cs
using System.Collections;

namespace Sort
{
    public class BubbleSort : ISort
    {
        public ArrayList Sort(ArrayList inArray)
        {
            for (int i = 0; i < inArray.Count; i++)
            {
                for (int j = i; j < inArray.Count; j++)
                {
                    if ((int)inArray[j] < (int)inArray[i])
                    {
                        var tmp = inArray[i];
                        inArray[i] = inArray[j];
                        inArray[j] = tmp;
                    }
                }
            }

            return inArray;
        }
    }
}

File: go/sort/bubbleSort.go
package sort

func BubleSort(inArray []int) []int {
    length := len(inArray)
    for i := 0; i < length-1; i++ {
        for j := i + 1; j < length; j++ {
            if inArray[j] < inArray[i] {
                inArray[i], inArray[j] = inArray[j], inArray[i]
            }
        }
    }

    return inArray
}
